### ISSUE TITLE
YAMon4 stops working after a while

### ISSUE THREAD

**Issue by richb-hanover** [al-caughey/YAMon-v4#21]
I apologize for this rather fuzzy/imprecise bug report but, in my experience, YAMon 4.0.7 simply stops working after a while (after a period of a couple months.) Furthermore, after a (possibly longer) period of time, the external USB drive stops working. Here's what I have seen...

* I installed 4.0.7 using the instructions from [usage-monitoring.com](http://usage-monitoring.com) I also use my own set of instructions for OpenWrt at: https://gist.github.com/richb-hanover/7e8b12d78c3de800afb1475789ac5595

* I have seen and fixed the symbolic link problem to make the web GUI work using the technique described at https://github.com/al-caughey/YAMon-v4/issues/1#issuecomment-615572947

* Then YAMon4 worked fine. It showed traffic patterns over the course of a day, week, etc.

* After a while, I forgot to check whether YAMon4 was working. But when I returned, the web GUI was not available at http://192.168.1.1/yamon/ Furthermore there were no YAMon processes running.

* Most surprising, the external USB flash drive I had installed/formatted/etc. was *not visible*. `ls -al /dev/sd*` did not show any device:

   ```
   root@~# ls -al /dev/sd*
   ls: /dev/sd*: No such file or directory
   ```
* And therefore, `/opt` (where I had installed YAMon) was gone.

This has happened with multiple (3-4) fresh installs of YAMon4 on at least two different OpenWrt routers running 18.06 or 19.07. 

Any thoughts on debugging this? Many thanks.

**Comment by alexandreloss** [comment]
Hi Rich, I confirm this issue too, running YAMON4 on a router with DD-WRT. The alternative workaround I found was checking hourly if YAMON is running and, if not, start the process.

**Comment by richb-hanover** [comment]
But have you ever observed the `/opt` partition disappear? Or the entire USB drive become non-functional/not seen?

**Comment by alexandreloss** [comment]
No, I haven't. /opt stays always up. In fact, my opt drive is mounted in an external SSD connected to a USB 3.0 port. My router is a Linkssys WRT3200 running DD-WRT (updated weekly to the latest release available).

**Comment by richb-hanover** [comment]
Curiouser and curiouser... My original post notes that `/opt` disappeared from the computer, and the USB drive wasn't visible in `ls`, and that certainly would cause YAMon to stop. 

Here's an update: I unplugged the USB drive from my OpenWrt router. When I inserted it into a Linux machine, it showed up and the files were as expected. install.sh and the YAMon4 directory were present at the top level, the lower-level files seemed to match what I expected. I didn't do a ton of troubleshooting. However, the latest `/opt/YAMon4/data/lastseen.js` file was dated 11 July... (I don't know if that's a hint or a red herring...)

I then plugged the USB back into the router (no change to any of the files), and the USB drive was immediately recognized. `ls -al /dev/sd*` and `ls /opt` showed the expected results.

A combination of `service yamon4 enable` and `service yamon4 start` and creating the symlink again, and maybe some other farbling around (I think I had to reboot to re-create `/tmp/www`) caused YAMon4 to begin running again. It has been fine for the last hour or so...

**My questions:** 

1. What might have caused YAMon to stop? 
2. What might have caused `/opt` to go away?
3. Might they be related? 

Two potential sources of trouble... I wrote *both* pieces of documentation that I used to get YAMon running. I'm not entirely certain that those procedures are correct, so it would be great to have someone else review them.

* I have created a new issue, requesting a review of my [Procedure for installing YAMon in OpenWrt](https://github.com/al-caughey/YAMon-v4/issues/22)
* It would be great to have someone look at my [Procedure for installing a USB drive in OpenWrt](https://openwrt.org/docs/guide-user/storage/usb-drives-quickstart)

**Comment by matlag** [comment]
Is it possible that your hard-drive draws more current that the router can provide, leading to a voltage drop and then a self power off? Maybe there's a trace in OpenWRT's logs of the exact time it turned off, so it would tell you if you have a single event triggering both events or if they are separated in time.

**Comment by richb-hanover** [comment]
That's an interesting thought. It's a USB 1GByte memory stick, so I wouldn't expect its power draw to be significant. BUT... I know more now than before:

1. It stopped working again on Tuesday after a power failure 
3. I ssh'd in after the power failure, `ls -al /dev/sd*` did not show any entries
4. Unplugging, waiting 5 seconds and re-plugging the USB stick didn't change - `ls ...` was empty
5. Rebooting the router didn't change anything - `ls ...` remained empty
6. Plugging the USB stick into a separate Unix box showed the expected directory structure which seemed OK. Files in YAMon4/data were last modified this Tuesday (power failure day)
7. After re-inserting it into the OpenWrt router, `ls /opt/YAMon4` showed files as expected
8. *(I also had trouble restarting YAMon, but will post that in a separate issue...)*

So, am I "doing something wrong" with this USB Drive? (I can't imagine that losing power makes all USB sticks fail for others...) I realize that this is a YAMon group, not a "help Rich debug his USB drive..." group - any thoughts about where to post these questions? Thanks.

**Update:** I posted this question to the [OpenWrt Forum](https://forum.openwrt.org/t/still-having-trouble-with-usb-drive/71196) - that's probably a better place to answer it... Thanks again.
